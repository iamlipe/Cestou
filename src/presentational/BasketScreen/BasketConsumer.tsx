import React from 'react';
import styled from 'styled-components/native';

export const BasketConsumer = () => {
  return (
    <StyledContainer>
      <StyledText>BasketConsumer</StyledText>
    </StyledContainer>
  );
};

const StyledContainer = styled.View``;
const StyledText = styled.Text``;
