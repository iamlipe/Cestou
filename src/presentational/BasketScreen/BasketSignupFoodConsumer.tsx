import React from 'react';
import styled from 'styled-components/native';

export const BasketSignupFoodConsumer = () => {
  return (
    <StyledContainer>
      <StyledText>BasketSignupFoodConsumer</StyledText>
    </StyledContainer>
  );
};

const StyledContainer = styled.View``;
const StyledText = styled.Text``;
