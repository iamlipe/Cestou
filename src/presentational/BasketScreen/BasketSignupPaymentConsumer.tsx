import React, {useState} from 'react';
import styled from 'styled-components/native';
import * as Yup from 'yup';
import {useForm} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {useGetConsumerBasket} from '@/hooks/useGetConsumerBasket';
import {useTranslation} from 'react-i18next';
import {getPixProducer} from '@/helpers/getPixProducer';
import {SvgProps} from 'react-native-svg';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {LoggedConsumerStackParamList} from '@/routes/stacks/LoggedConsumerStack';
import {useNavigation} from '@react-navigation/native';
import {translateBasketToPortuguese} from '@/helpers/translate';

import BasketVegetable from '@/assets/svgs/vegetable-basket.svg';

import Header from '@/components/Header';
import RadioForm from '@/components/RadioForm';
import Button from '@/components/Button';
import InputClipboard from '@/presentational/BasketScreen/InputClipboard';
import Modal from '@/components/Modal';

import {
  StyledContainerScroll,
  StyledContent,
  StyledText,
} from './BasketSignupPlanConsumer';

interface Form {
  typeDeleviry: string;
}

type NavProps = NativeStackNavigationProp<
  LoggedConsumerStackParamList,
  'ProfileConsumer' | 'DonationConsumerStack'
>;

export const BasketSignupPaymentConsumer = () => {
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  const {navigate} = useNavigation<NavProps>();
  const {t, i18n} = useTranslation();
  const {basketConsumer} = useGetConsumerBasket();

  const schema = Yup.object().shape({
    typeDeleviry: Yup.string().required(t('error.required')),
  });

  const {
    control,
    handleSubmit,
    formState: {errors, isSubmitted},
  } = useForm<Form>({resolver: yupResolver(schema)});

  function handleModal(event: boolean) {
    setIsVisibleModal(event);
  }

  function handleNavigateToProfile() {
    handleModal(false);
    navigate('ProfileConsumer');
  }

  function handleNavigateToDonations() {
    handleModal(false);
    navigate('DonationConsumerStack');
  }

  async function onSubmit() {
    handleModal(true);
  }

  return (
    <StyledContainerScroll
      showsVerticalScrollIndicator={false}
      testID="basket-signup-payment-consumer-screen">
      <Header
        title={t('text.basketSignupPaymentConsumer.headerTitle')}
        welcome={false}
      />
      <StyledContent>
        <StyledTitle>
          {t('text.basketSignupPaymentConsumer.titleSectionOne')}
        </StyledTitle>
        <StyledText>
          {t('text.basketSignupPaymentConsumer.instructionSectionOne')}
        </StyledText>

        <RadioForm
          name="typeDeleviry"
          control={control}
          options={[
            t('option.typeDeliveryCollectPoint'),
            t('option.typeDeliveryHome'),
          ]}
          type="withLine"
          error={isSubmitted ? errors.typeDeleviry?.message : ''}
        />

        <StyledCheckoutBox>
          <StyledTitle>
            {t('text.basketSignupPaymentConsumer.checkoutTitle')}
          </StyledTitle>
          <StyledRow>
            {basketConsumer?.basketID.size && (
              <StyledText>
                {t(
                  'text.basketSignupPaymentConsumer.checkoutDescriptionBasket',
                  {
                    size:
                      i18n.language === 'pt'
                        ? translateBasketToPortuguese(
                            basketConsumer?.basketID.size,
                          )
                        : basketConsumer?.basketID.size,
                  },
                )}
              </StyledText>
            )}

            <StyledText>
              {t('text.basketSignupPaymentConsumer.checkoutPriceBasket', {
                price: basketConsumer?.basketID.value,
              })}
            </StyledText>
          </StyledRow>
          <StyledRow>
            <StyledText>
              {t('text.basketSignupPaymentConsumer.checkoutTotal')}
            </StyledText>
            <StyledTextTotalPrice>
              {t('text.basketSignupPaymentConsumer.checkoutTotalPriceBasket', {
                price: basketConsumer?.basketID.value,
              })}
            </StyledTextTotalPrice>
          </StyledRow>
        </StyledCheckoutBox>

        <StyledTitle>
          {t('text.basketSignupPaymentConsumer.titleSectionTwo')}
        </StyledTitle>

        <StyledText>
          {t('text.basketSignupPaymentConsumer.instructionSectionTwo')}
        </StyledText>

        {basketConsumer &&
          (basketConsumer?.basketProducerID.randomPix ||
            basketConsumer?.basketProducerID.phonePix ||
            basketConsumer?.basketProducerID.emailPix ||
            basketConsumer?.basketProducerID.cpfPix) && (
            <InputClipboard
              text={getPixProducer(basketConsumer?.basketProducerID)}
            />
          )}

        <StyledSubmitButton
          testID="submit-button-payment"
          title={t('button.completePurchase')}
          onPress={handleSubmit(onSubmit)}
        />

        {isVisibleModal && (
          <Modal
            title={t('modal.titleCompletePurchase')}
            subTitle={t('modal.subTileCompletePurchase')}
            icon={BasketVegetable as React.FC<SvgProps>}
            titleConfirmButton={t('button.goToMessages')}
            titleCancelButton={t('button.goToDonations')}
            columnButtons
            onConfirm={handleNavigateToProfile}
            onCancel={handleNavigateToDonations}
            onClose={() => handleModal(false)}
          />
        )}
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGER};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin-bottom: 16px;
`;

const StyledCheckoutBox = styled.View`
  height: 146px;
  justify-content: center;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_800};
  border-radius: 4px;
  padding: 16px;
  margin-bottom: 16px;
`;

const StyledRow = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const StyledTextTotalPrice = styled(StyledText)`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
`;

const StyledSubmitButton = styled(Button)`
  margin-top: 24px;
`;
