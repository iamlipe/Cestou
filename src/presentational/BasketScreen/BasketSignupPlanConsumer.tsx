import React, {useEffect} from 'react';
import styled from 'styled-components/native';
import * as Yup from 'yup';
import {useForm} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {useReduxSelector} from '@/hooks/useReduxSelector';
import {useReduxDispatch} from '@/hooks/useReduxDispatch';
import {GET_BASKET} from '@/store/slices/basketSlice';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {BasketConsumerStackParamList} from '@/routes/stacks/BasketConsumerStack';

import Header from '@/components/Header';
import RadioForm from '@/components/RadioForm';
import Button from '@/components/Button';

const schema = Yup.object().shape({
  daysPerDeliver: Yup.string().required('Preenchimento obrigatório'),
  size: Yup.string().required('Preenchimento obrigatório'),
});

interface SubmitForm {
  daysPerDeliver: string;
  size: string;
}

type NavProps = NativeStackNavigationProp<
  BasketConsumerStackParamList,
  'BasketSignupFoodConsumer'
>;

export const BasketSignupPlanConsumer = () => {
  const {navigate} = useNavigation<NavProps>();

  const {
    control,
    handleSubmit,
    formState: {errors, isSubmitted},
  } = useForm<SubmitForm>({resolver: yupResolver(schema)});

  function handleNavigate(data: SubmitForm) {
    navigate('BasketSignupFoodConsumer', {...data});
  }

  async function onSubmit(data: SubmitForm) {
    handleNavigate(data);
  }

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header title="Assinatura" welcome={false} />
      <StyledContent>
        <StyledTitle>Vamos configurar seu plano</StyledTitle>
        <StyledLabel>
          Escolha qual a frequência de recebimento ou retirada da sua cesta:
        </StyledLabel>
        <RadioForm
          name="daysPerDeliver"
          control={control}
          options={['Semanal', 'Quinzenal']}
          type="withBox"
          error={isSubmitted ? errors.daysPerDeliver?.message : ''}
        />

        <StyledLabel>
          Escolha qual o tamanho deseja para sua cesta de alimentos:
        </StyledLabel>

        <StyledLine />
        <RadioForm
          name="size"
          control={control}
          options={[
            'Pequena - R$ 45,00',
            'Média - R$ 70,00',
            'Grande - R$90,00',
          ]}
          detailsOptions={[
            '1 tempero, 2 legumes, 2 verduras , 3 frutas',
            '2 temperos, 3 legumes, 3 verduras, 3 frutas e 1 processado',
            '3 temperos, 4 legumes, 4 verduras, 4 frutas e 1 processado',
          ]}
          type="withBox"
          error={isSubmitted ? errors.size?.message : ''}
        />
        <StyledText>
          Obs: O valor informado acima é cobrado de acordo com a frequência
          escolhida.
        </StyledText>

        <StyledSubmitButton title="Próximo" onPress={handleSubmit(onSubmit)} />
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGEST};
  color: ${({theme}) => theme.colors.GRAY_900};
  text-align: center;
  margin-bottom: 24px;
`;

const StyledLabel = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin-bottom: 16px;
`;

const StyledText = styled(StyledLabel)``;

const StyledLine = styled.View`
  width: 100%;
  height: 1px;
  align-self: center;
  background-color: ${({theme}) => theme.colors.GRAY_100};
  margin-bottom: 16px;
`;

const StyledSubmitButton = styled(Button)``;
