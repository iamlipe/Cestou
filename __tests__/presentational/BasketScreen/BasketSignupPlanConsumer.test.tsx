import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';
import {fireEvent} from '@testing-library/react-native';
import {act} from 'react-test-renderer';

import {BasketSignupPlanConsumer} from '@/presentational/BasketScreen/BasketSignupPlanConsumer';

describe('BasketSignupPlanConsumer', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<BasketSignupPlanConsumer />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<BasketSignupPlanConsumer />);
  });

  test('should form submit', async () => {
    const {getAllByTestId, getByTestId, getByText} = renderWithThemeProvider(
      <BasketSignupPlanConsumer />,
    );

    const optionDaysPerDeliver = getAllByTestId('radio-icon-not-checked-1')[0];
    const optionSize = getAllByTestId('radio-icon-not-checked-0')[1];
    const buttonSubmit = getByText(/prÃ³ximo/i);

    fireEvent.press(optionDaysPerDeliver);
    fireEvent.press(optionSize);

    await act(() => fireEvent.press(buttonSubmit));

    const optionDaysPerDeliverChecked = getByTestId('radio-icon-checked-1');
    const optionSizeChecked = getByTestId('radio-icon-checked-0');

    expect(optionDaysPerDeliverChecked).toBeTruthy();
    expect(optionSizeChecked).toBeTruthy();
  });
});
