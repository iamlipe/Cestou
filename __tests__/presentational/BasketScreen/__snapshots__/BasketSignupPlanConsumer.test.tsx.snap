// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketSignupPlanConsumer should matches snapshot 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "minHeight": "100%",
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6CCD91",
              "flexDirection": "row",
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": "5%",
              "paddingRight": "5%",
              "paddingTop": 0,
              "width": "100%",
            },
          ]
        }
        type="normal"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "space-between",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#1A1A1A",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 47,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Assinatura
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": 24,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 24,
            "width": "90%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 47,
              "marginBottom": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Vamos configurar seu plano
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 27,
              "marginBottom": 16,
            },
          ]
        }
      >
        Escolha qual a frequência de recebimento ou retirada da sua cesta:
      </Text>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "maxHeight": 80,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#3A3A3A",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="radio-icon-not-checked-0"
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              detailsOptions={false}
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 27,
                    "marginLeft": 8,
                  },
                ]
              }
            >
              Semanal
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "maxHeight": 80,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#3A3A3A",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="radio-icon-not-checked-1"
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              detailsOptions={false}
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 27,
                    "marginLeft": 8,
                  },
                ]
              }
            >
              Quinzenal
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#781B1C",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 24,
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": -8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 27,
              "marginBottom": 16,
            },
          ]
        }
      >
        Escolha qual o tamanho deseja para sua cesta de alimentos:
      </Text>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "#F2F2F8",
              "height": 1,
              "marginBottom": 16,
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "maxHeight": 80,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#3A3A3A",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="radio-icon-not-checked-0"
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              detailsOptions={true}
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 27,
                    "marginLeft": 8,
                  },
                ]
              }
            >
              Pequena
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 27,
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                ]
              }
            >
              1 tempero, 2 legumes, 2 verduras , 3 frutas
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "maxHeight": 80,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#3A3A3A",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="radio-icon-not-checked-1"
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              detailsOptions={true}
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 27,
                    "marginLeft": 8,
                  },
                ]
              }
            >
              Média
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 27,
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                ]
              }
            >
              2 temperos, 3 legumes, 3 verduras, 3 frutas
e 1 processado
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "maxHeight": 80,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#3A3A3A",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="radio-icon-not-checked-2"
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              detailsOptions={true}
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 27,
                    "marginLeft": 8,
                  },
                ]
              }
            >
              Grande
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 27,
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                ]
              }
            >
              3 temperos, 4 legumes, 4 verduras, 4 frutas
e 1 processado
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#781B1C",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 24,
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": -8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 27,
              "marginBottom": 16,
            },
          ]
        }
      >
        Obs: O valor informado acima é cobrado de acordo com a frequência escolhida.
      </Text>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#00843F",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 49,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
            "width": 200,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 27,
              },
            ]
          }
          textColor="white"
        >
          Próximo
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
